Sources/Commands/SwiftPackageTool.swift

```diff
@@ -969,6 +978,31 @@ extension SwiftPackageTool {
             // FIXME: Revisit this path.
             let outputDir = pluginsDir.appending(component: "outputs")
 
+            // Determine the set of directories under which plugins are allowed to write. We always include the output directory.
+            var writableDirectories = [outputDir]
+            if allowWritingToPackageDirectory {
+                writableDirectories.append(package.path)
+            }
+            if !blockWritingToTemporaryDirectory {
+                writableDirectories.append(AbsolutePath("/tmp"))
+                writableDirectories.append(AbsolutePath(NSTemporaryDirectory()))
```

Comment:
I think we recently added tempDirectory to FileSystem, maybe we can use it here?